components/match2/wikis/brawlstars/legacy/match_maps_legacy.lua

```diff
@@ -0,0 +1,329 @@
+---
+-- @Liquipedia
+-- wiki=brawlstars
+-- page=Module:MatchMaps/Legacy
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local CharacterIcon = require('Module:CharacterIcon')
+local Arguments = require('Module:Arguments')
+local Array = require('Module:Array')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Json = require('Module:Json')
+local MatchGroup = require('Module:MatchGroup')
+local PageVariableNamespace = require('Module:PageVariableNamespace')
+local Table = require('Module:Table')
+local Template = require('Module:Template')
+
+local CharacterNames = mw.loadData('Module:BrawlerNames')
+
+local MatchGroupBase = Lua.import('Module:MatchGroup/Base')
+local MatchSubobjects = Lua.import('Module:Match/Subobjects')
+
+local globalVars = PageVariableNamespace()
+local matchlistVars = PageVariableNamespace('LegacyMatchlist')
+
+local MAX_NUMBER_OF_OPPONENTS = 2
+local DEFAULT_WIN = 'W'
+local DEFAULT_LOSS = 'L'
+local FORFEIT = 'FF'
+local TBD = 'tbd'
+
+local MatchMapsLegacy = {}
+
+---@param args table
+---@return table
+function MatchMapsLegacy._handleMaps(args)
+	Array.mapIndexes(function (index)
+		local prefix = 'map' .. index
+		local map = args[prefix]
+		local winner = Table.extract(args, prefix .. 'win')
+		if Logic.isEmpty(map) then
+			return false
+		end
+		local score = Table.extract(args, prefix .. 'score')
+		local score1
+		local score2
+		if Logic.isNotEmpty(score) then
+			local splitedScore = mw.text.split(score, '-')
+			score1 = splitedScore[1]
+			score2 = splitedScore[2]
+		end
+
+		args[prefix .. 'score1'] = mw.text.trim(score1 or '')
+		args[prefix .. 'score2'] = mw.text.trim(score2 or '')
+		args[prefix .. 'winner'] = winner
+		args[prefix .. 'maptype'] = Table.extract(args, prefix ..'type')
+
+		return true
+	end)
+
+	return args
+end
+
+---@param args table
+---@return table
+function MatchMapsLegacy._handleBans(args)
+	local bans = {}
+	Array.mapIndexes(function (index)
+		local ban = Table.extract(args, 'ban' .. (index == 1 and '' or index))
+		if Logic.isEmpty(ban) then
+			return false
+		end
+		table.insert(bans, CharacterIcon.Icon({
+			character = CharacterNames[ban:lower()],
+			size = '30px'
+		}))
+		return true
+	end)
+
+	if #bans == 0 then
+		return args
+	end
+
+	local bansComment = '\'\'\'Bans\'\'\':' .. table.concat(bans, '&nbsp;')
+	args.comment = args.comment and (args.comment .. '<br>' .. bansComment) or bansComment
+	return args
+end
```

Comment:
Maybe we can handle this via match group input, where bans are input for the whole match and not map specific, but I will leave it as it is now.